halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/kubernetes/KubernetesAccountValidator.java

```diff
@@ -0,0 +1,156 @@
+      parent = parent.getParent();
+    }
+    deploymentConfiguration = (DeploymentConfiguration) parent;
+
+    validateDockerRegistries(psBuilder, account, deploymentConfiguration);
+    validateKubeconfig(psBuilder, account);
+  }
+
+  private void validateKubeconfig(ProblemSetBuilder psBuilder, KubernetesAccount account) {
+    io.fabric8.kubernetes.api.model.Config kubeconfig;
+    String context = account.getContext();
+    String kubeconfigFile = account.getKubeconfigFile();
+    String cluster = account.getCluster();
+    String user = account.getUser() ;
+    List<String> namespaces = account.getNamespaces();
+
+    // This indicates if a first pass at the config looks OK. If we don't see any serious problems, we'll do one last check
+    // against the requested kubernetes cluster to ensure that we can run spinnaker.
+    boolean smoketest = true;
+
+    try {
+      File kubeconfigFileOpen = new File(kubeconfigFile);
+      kubeconfig = KubeConfigUtils.parseConfig(kubeconfigFileOpen);
+    } catch (IOException e) {
+      psBuilder.addProblem(ERROR, e.getMessage());
+      return;
+    }
+
+    if (context != null && !context.isEmpty()) {
+      Optional<NamedContext> namedContext = kubeconfig
+          .getContexts()
+          .stream()
+          .filter(c -> c.getName().equals(context))
+          .findFirst();
+
+      if (!namedContext.isPresent()) {
+        psBuilder.addProblem(ERROR, "Context \"" + context + "\" not found in kubeconfig \"" + kubeconfigFile + "\"")
+            .setRemediation("Either add this context to your kubeconfig, rely on the default context, or pick another kubeconfig file");
+        smoketest = false;
+      }
+    } else {
+      String currentContext = kubeconfig.getCurrentContext();
+      if (currentContext.isEmpty()) {
+        psBuilder.addProblem(ERROR, "You have not specified a Kubernetes context, and your kubeconfig \"" + kubeconfigFile + "\" has no current-context")
+            .setRemediation("Either specify a context in your halconfig, or set a current-context in your kubeconfig");
+        smoketest = false;
+      } else {
+        psBuilder.addProblem(WARNING, "You have not specified a Kubernetes context in your halconfig, Spinnaker will use \"" + kubeconfigFile + "\" instead")
+            .setRemediation("We recommend explicitly setting a context in your halconfig, to ensure changes to your kubeconfig won't break your deployment");
+      }
+    }
+
+    if (smoketest) {
+      Config config = KubernetesConfigParser.parse(kubeconfigFile, context, cluster, user, namespaces);
+      KubernetesClient client = new DefaultKubernetesClient(config);
+
+      try {
+        client.namespaces().list();
+      } catch (Exception e) {
+        psBuilder.addProblem(ERROR, "Unable to communicate with your Kubernetes cluster: " + e.getMessage())
+            .setRemediation("Verify that these credentials work manually using \"kubectl\"");
```

Comment:
"these credentials" which are "these" or is that clear from the reference reported with the remediation?